charactermeth.cpp

```diff
@@ -1,38 +1,41 @@
 
-#include <iostream>

```

Comment:
Az jó, hogy innen kikerült az iostream, hisz már nincs használva, de a headerben meg nem is volt használva, úgyhogy ott se legyen include-olva!

Reply:
A game.h-ban van az #include <iostream>. Habár ott konkrétan nincs használva viszont a main.cpp illetve a gamemeth.cpp -ben kell. Ha nem a game.h-ban hivatkozom akkor mind a két cpp file-ba kell az #include <iostream>. Így viszont elég egy helyen a game.h-ban

Reply:
Nem az a lényeg, hogy kevesebbszer kelljen leírni, hanem hogy csak ott legyen include-olva, ahol használva is van. A header fájlok (mint a character.h) sok helyen be lehetnek include-olva, ezért azokban minél kevesebb ilyen felesleges en beinclude-olt kód legyen. A cpp-k viszont mind csak egyszer lesznek lefordítva, és ott úgyse lehet elkerülni azt az include-ot, ami ott használva van.

Reply:
Rendben, erre így nem gondoltam. Köszönöm.

Reply:
Szívesen. Majd a `make` témakörnél lesz erről bővebben szó, meg a források újrafordításáról.